lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/MapTaskHost.cs

```diff
@@ -59,15 +60,16 @@ private MapTaskHost(IMapFunction<TMapInput, TMapOutput> mapTask, IGroupCommClien
         /// <returns></returns>
         public byte[] Call(byte[] memento)
         {
-            MapInputWithControlMessage<TMapInput> mapInput = _dataAndMessageReceiver.Receive();
-
-            while (mapInput.ControlMessage == MapControlMessage.AnotherRound)
+            while (true)
             {
+                GC.Collect();
```

Comment:
@anupam128 

Regarding your points:

1) Agreed and done
2) Regarding JobObject: I do not understand it completely. But after @afchung  comments above, do you still think its a good idea. If yes then I will look in to it and then take your help.
3) I think yarn RM is killing it. @markusweimer please confirm.
4) Yes, it is possible to some extent. In group communication there is trade-off between memory generated (for example for pipelining and all) vs. speed of communication. Making it memory efficient while maintaining speed is extremely tricky involving things like maintaining buffers or caches, getting acknowledgement messages back from the nodes to which you are sending messages which will increase the communication latencies (might not matter for large dimensional communications which are actual source of large memory usage) and other complications. So they can be done but as a long term well thought out solution rather than some immediate things. However, its definitely worth  looking at after first round.